### PR TITLE
Allow the wstat statistic to be used in the liklihood-ratio test

### LINKED ISSUE #1745: can I use pvalue plot with wstat?
I have

```python
np.random.seed(3287)
ui.set_stat("wstat")
print(f"stat = {ui.get_stat_name()}")
ui.get_pvalue_plot(null_model=null,
                   alt_model=alt,
                   conv_model=rsp,
                   num=5, numcores=1, recalc=True)
```

and this errors out with

```
stat = wstat
Traceback (most recent call last):
  File "/lagado.real/sherpa/sherpa-main/sherpa-test-data/sherpatest/testy.py", line 18, in <module>
    ui.get_pvalue_plot(null_model=null,
  File "<string>", line 1, in get_pvalue_plot
  File "/lagado2.real/local/anaconda/envs/ciao415-py310/lib/python3.10/site-packages/sherpa/astro/ui/utils.py", line 10704, in get_pvalue_plot
    return super().get_pvalue_plot(null_model=null_model, alt_model=alt_model,
  File "/lagado2.real/local/anaconda/envs/ciao415-py310/lib/python3.10/site-packages/sherpa/ui/utils.py", line 9173, in get_pvalue_plot
    results = pvalue(fit, null_model, alt_model,
  File "/lagado2.real/local/anaconda/envs/ciao415-py310/lib/python3.10/site-packages/sherpa/sim/simulate.py", line 233, in run
    raise TypeError("Sherpa fit statistic must be Cash or CStat" +
TypeError: Sherpa fit statistic must be Cash or CStat for likelihood ratio test
```

I would have thought we could use wstat here.

There's a larger issue here about how Sherpa gates certain functionality - in this case it's with

```python
        if not isinstance(stat, (Cash, CStat)):
            raise TypeError("Sherpa fit statistic must be Cash or CStat" +
                            " for likelihood ratio test")
```

that I have an ongoing low-level desire to improve, but there's no simple golden fix, so we need to treat each case we find.

### PR DESCRIPTION
**Pull request by DougBurke**
# Summary

Allow the Likelihood Ratio Test (as used by `plot_value`) to be run when using the WStat statistic. Fix #1745.

# Details

Is it just as easy as not raising an error if given a WStat statistc?

The tests are awkward to write here, and I'm basically just checking we get repeatable behaviour (as much as possible, which unfortunately isn't much for the ratio values given the RNG changes I made in #1735, in particular the `parallel_map_rng` routine).

### CODE REVIEW

**Comment by anetasie** [comment]
@DougBurke I just tried `plot_pvalue` with `wstat` in this branch and it run without an error and generated a reasonable values.
Are you ready to get this into the release?

**Comment by DougBurke** [comment]
It's too late.  I haven't looked at this for a while and I don't know what
is still needed.

**Review by anetasie — APPROVED** [review]

**Comment by anetasie** [comment]
@DougBurke I did run my tests and this looks good. Thanks!

File: sherpa/sim/simulate.py [hunk 1]
```diff
@@ -25,24 +25,27 @@
...
 from sherpa.fit import Fit
+from sherpa.plot import arr2str
 from sherpa.sim.sample import NormalParameterSampleFromScaleMatrix
 from sherpa.utils import NoNewAttributesAfterInit
 from sherpa.utils.parallel import parallel_map_rng
 from sherpa.utils.random import poisson_noise
+from sherpa.utils.types import ArrayType
 
 
+# TODO: this should probaly use __name__ rather than "sherpa"
```
Comment (wmclaugh) [inline sherpa/sim/simulate.py:44]:
```suggestion
# TODO: this should probably use __name__ rather than "sherpa"
```

**Pull request merged**